librz/include/rz_util/rz_print.h

```diff
@@ -140,7 +140,7 @@ RZ_API void rz_print_set_flags(RzPrint *p, int _flags);
 RZ_API void rz_print_addr(RzPrint *p, ut64 addr);
 RZ_API void rz_print_section(RzPrint *p, ut64 at);
 RZ_API void rz_print_hexii(RzPrint *p, ut64 addr, const ut8 *buf, int len, int step);
-RZ_API void rz_print_hexdump(RzPrint *p, ut64 addr, const ut8 *buf, int len, int base, int step, size_t zoomsz);
+RZ_API RZ_OWN char *rz_print_hexdump_str(RZ_NONNULL RzPrint *p, ut64 addr, RZ_NONNULL const ut8 *buf, int len, int base, int step, size_t zoomsz);
```

Comment:
i like the changes, but i would prefer to move `rz_print_hexdump_str` under rz_core and just have something like
```c
RZ_API void rz_core_hexdump_block(RZ_NONNULL RzCore *core, int base, int step, size_t zoomsz);
```
the function should use directly `core->offset` and `core->block` otherwise put it under rz_cons as this
```c
RZ_API void rz_cons_hexdump(ut64 addr, RZ_NONNULL const ut8 *buf, int len, int base, int step, size_t zoomsz);
```
@XVilka @ret2libc what do you think?

Reply:
Yes, but printing should not be in the API - it's useless if you use it as a library. String is more useful

Reply:
Fine to move this into the Core, just please return string

Reply:
I prefer avoiding direct use of core->offset and core->block in APIs because they require users of the api to call seek_set and restore the seek after the call, which is painful. It is much easier to use an API that automatically reads the data it needs from `addr`, passed as argument. Also, core->block is just a block of data and maybe the API needs more, so it is in general bad to use IMHO. AFAIR we are trying to move away from core->block in favour of rz_io_read_at.

Reply:
About Core, I'm ok with keeping *this* function in RzPrint and add `rz_core_hexdump_block` as suggested by @wargio , which simply calls rz_print_hexdump_str and print it.

Reply:
:S i actually want to remove completely RzPrint, so this either goes to rz_cons or rz_core for me.

Reply:
Since the others are saying to do the opposite as me, then just follow what @ret2libc said, just add an extra method to dump the core->block data.

Reply:
I'm not saying RzPrint is good actually :D I also think it is a bit useless to have such a thing and probably better to have everything in RzCore, but... let's do that at some point later I guess.